caput/mpiarray.py

```diff
@@ -807,3 +840,107 @@ def _to_hdf5_serial(self, filename, dataset, create=False):
                     fh[dataset][start:end] = dist_arr
 
             dist_arr.comm.Barrier()
+
+    def _partition_io(self, skip=False, threshold=1.99):
+        """Split IO of this array into local sections under `threshold`.
+
+        Parameters
+        ----------
+        skip : bool, optional
+            Don't partition, just find and return a full axis.
+        threshold : float, optional
+            Maximum size of IO (in GB).
+
+        Returns
+        -------
+        split_axis : int
+            Which axis are we going to split along.
+        partitions : list of slice objects
+            List of slices.
+        """
+        from mpi4py import MPI
+
+        threshold_bytes = threshold * 2**30
+        largest_size = self.comm.allreduce(self.nbytes, op=MPI.MAX)
+        num_split = int(np.ceil(largest_size / threshold_bytes))
+
+        if skip or self.ndim == 1:
```

Comment:
should this be `if skip or num_split == 1:` ? Seems like if you are not splitting up the IO you can just return at this point. What about cases where the array is 1D but greater than 2GB?

Reply:
The issue here is that it can only split across a different axis to the distributed axis, so if there's only one axis it can't do anything. Also, if `num_split == 1` that's a totally normal circumstance to be in (it just means the array is small enough it doesn't need to be split). You are right though that if num_split does equal one then we could exit at this point. I can try and change the logic around to make that clear, also to warn if there is only axis which is too large.

Reply:
Yeah, that makes sense. Does it cause an mpi-io issue if there is only one axis and it is over 2GB? That should be a pretty rare case for us, but maybe a warning could help track that down if it ever comes up.